pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -60,8 +62,9 @@ func (e *endpointsController) registerEndpointsHandler() {
 				// Avoid pushes if only resource version changed (kube-scheduller, cluster-autoscaller, etc)
 				oldE := old.(*v1.Endpoints)
 				curE := cur.(*v1.Endpoints)
+				if oldE.ResourceVersion != curE.ResourceVersion && !compareEndpoints(oldE, curE) {
```

Comment:
for my own education, who sets this resource version on old and new? If it is the one from https://github.com/istio/istio/pull/19844/files#diff-714b1e49afcb2c6464187f698c680635R1300, then wont it always be different because its effectively generating a random UUID for every endpoint object?

Reply:
In real env, it is kube-apiserver who set the resource version. Please see the pr description.